lib/note-detail.jsx

```diff
@@ -34,22 +30,34 @@ export default React.createClass( {
 		return note && note.id;
 	},
 
-	componentWillReceiveProps: function( { note } ) {
-		this.setState( {
-			content: get( note, 'data.content', '' )
-		} );
+	componentWillReceiveProps: function() {
+		const { note } = this.props;
+
+		if ( ! this.isValidNote( note ) || ! this.noteEditor ) return;
+
+		if ( saveTimerId ) clearTimeout( saveTimerId );
+
+		// Save note if content changes haven't been saved yet
+		if ( note.data.content !== this.noteEditor.value ) {
+			this.props.onChangeContent( note, this.noteEditor.value );
+			analytics.tracks.recordEvent( 'editor_note_edited' );
+		}
 	},
 
 	componentDidUpdate: function() {
-		// Let's focus the editor for new and empty notes
 		const { note } = this.props;
-		const { content } = this.state;
-		if ( this.isValidNote( note ) && content === '' ) {
-			this.noteEditor.focus();
+		const content = get( note, 'data.content', '' );
+		if ( this.isValidNote( note ) && this.noteEditor ) {
+			this.noteEditor.value = content;
+
+			// Let's focus the editor for new and empty notes
+			if ( content === '' ) {
+				this.noteEditor.focus();
+			}
 		}
 	},
 
-	onPreviewClick( event ) {
+	onPreviewClick: function( event ) {
```

Comment:
Yeah, even though it's cool these days to not do that, I was matching it up with the rest of the class ;)